packages/medusa/src/types/global.ts

```diff
@@ -37,10 +37,38 @@ export type Logger = _Logger & {
   warn: (msg: string) => void
 }
 
+export enum MODULE_SCOPE {
+  INTERNAL = "internal",
+  EXTERNAL = "external",
+}
+
+export enum MODULE_RESOURCE_TYPE {
+  SHARED = "shared",
+  ISOLATED = "isolated",
+}
+
+export type ConfigurableModuleDeclaration = {
+  scope: MODULE_SCOPE.INTERNAL
+  resources: MODULE_RESOURCE_TYPE
```

Comment:
suggestion: should we set it to shared as of now?

Reply:
I think is OK to keep it here. The 2 modules we have aren't supporting isolated resources for now, but that doesn't mean others couldn't.